### PR TITLE
on login page ignore security fixes #222

### LINKED ISSUE #222: Rotating private/public keys causes login page to Internal Server Error
If you rotate the private/public keys of a Renarde application then when you visit directly the login page it causes an Internal Server Error by `RenardeSecurity#makeRedirectToLogin`.

```java
if (request.uri().equals(config.getLoginPage())) {
            // this would cause a redirect loop, not sure how to handle, but not by redirecting
            log.errorf("Redirect loop at %s, giving up on clearing bad JWT cookie", config.getLoginPage());
            return Response.serverError().build();
}
```

### PR DESCRIPTION
**Pull request by gbourant**

### CODE REVIEW

**Comment by gbourant** [comment]
If this is about to be added i will add some tests and also investigate to remove some code which is used to avoid `redirect loop`

**Comment by FroMage** [comment]
Can you explain why this is needed and what triggers it? Perhaps with a failing test? I'm surprised I didn't run into this earlier.

**Comment by gbourant** [comment]
Also my fix is wrong because it makes `SecurityIdentity#isAnonymous` to misbehave in the login page. It always returns false even when the user is already logged in.

**Comment by FroMage** [comment]
OK. So this happens when trying to access the login page with an invalid cookie. Good point. I've always tested accessing a regular page with an invalid cookie, which would redirect me to the login page after clearing the invalid cookie.

I suppose in this case, it's fine to redirect from the login page to the login page, once, simply because we're clearing the cookie. So it's not a loop. Perhaps add a check in `RenardeSecurity.makeRedirectToLogin` to allow for redirecting to the login page if we have a cookie to clear?

**Comment by gbourant** [comment]
I've added the check to the `RenardeSecurity.makeRedirectToLogin` method.

**Comment by FroMage** [comment]
OK that looks good. Could you add your test please? Probably in `JWTTest` in `quarkus-renarde/deployment`.

**Comment by gbourant** [comment]
I just pushed the test. For some weird reason when i put the test in `JWTTest` class i get the following error, so i added it to the `CustomLoginControllerTest` class if that's ok.

```
[ERROR] Errors: 
[ERROR]   JWTTest » Runtime java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkiverse.renarde.deployment.RenardeProcessor#collectControllers threw an exception: java.lang.IllegalArgumentException: Cannot provide/consume multiple values for class io.quarkiverse.renarde.deployment.LoginPageBuildItem
```

File: deployment/src/test/java/io/quarkiverse/renarde/test/CustomLoginControllerTest.java [hunk 1]
```diff
@@ -78,6 +83,41 @@ public void testSuccessfulLoginPage() {
...
+                .innerSign().encrypt();
+
+        var response = given()
+                .redirects().follow(false)
+                .when()
+                .cookie("QuarkusUser", token)
+                .get("/login").then()
+                .statusCode(303)
+                .extract().response();
+        assertFalse(response.headers().hasHeaderWithName("QuarkusUser"));
```
Comment (FroMage) [inline deployment/src/test/java/io/quarkiverse/renarde/test/CustomLoginControllerTest.java:118]: This is not testing that it clears the cookie, it checks for headers :)

File: deployment/src/test/java/io/quarkiverse/renarde/test/CustomLoginControllerTest.java [hunk 2]
```diff
@@ -78,6 +83,41 @@ public void testSuccessfulLoginPage() {
...
+                .upn("user")
+                .innerSign().encrypt();
+
+        var response = given()
+                .redirects().follow(false)
+                .when()
+                .cookie("QuarkusUser", token)
+                .get("/login").then()
+                .statusCode(303)
+                .extract().response();
```
Comment (FroMage) [inline deployment/src/test/java/io/quarkiverse/renarde/test/CustomLoginControllerTest.java:117]: You should also check that you're redirected to `/login`

**Review by FroMage — CHANGES_REQUESTED** [review]

**Review by FroMage — APPROVED** [review]
Excellent, thanks!

**Pull request merged**